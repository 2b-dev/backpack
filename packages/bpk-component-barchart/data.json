{
  "continentCountries": [
    {
      "continent": "North America",
      "countries": 3
    },
    {
      "continent": "South America",
      "countries": 13
    },
    {
      "continent": "Europe",
      "countries": 47
    },
    {
      "continent": "Africa",
      "countries": 44
    },
    {
      "continent": "Asia",
      "countries": 50
    },
    {
      "continent": "Australia",
      "countries": 1
    },
    {
      "continent": "Antartica",
      "countries": 0
    }
  ],
  "example1": [
    {
      "value": 11,
      "label": "1"
    },
    {
      "value": 13,
      "label": "2"
    },
    {
      "value": 5,
      "label": "3"
    },
    {
      "value": 16,
      "label": "4"
    },
    {
      "value": 4,
      "label": "5"
    },
    {
      "value": 19,
      "label": "6"
    },
    {
      "value": 10,
      "label": "7"
    },
    {
      "value": 5,
      "label": "8"
    },
    {
      "value": 8,
      "label": "9"
    },
    {
      "value": 3,
      "label": "10"
    }
  ],
  "example2": [
    {
      "value": 11,
      "label": "1"
    },
    {
      "value": 13,
      "label": "2"
    },
    {
      "value": 75,
      "label": "3"
    },
    {
      "value": 16,
      "label": "4"
    },
    {
      "value": 4,
      "label": "5"
    },
    {
      "value": 19,
      "label": "6"
    },
    {
      "value": 45,
      "label": "7"
    },
    {
      "value": 5,
      "label": "8"
    },
    {
      "value": 8,
      "label": "9"
    },
    {
      "value": 3,
      "label": "10"
    }
  ]
}
