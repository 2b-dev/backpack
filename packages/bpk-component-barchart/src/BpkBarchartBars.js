import React from 'react';
import PropTypes from 'prop-types';
import { path as d3Path } from 'd3-path';
import { durationBase } from 'bpk-tokens/tokens/base.es6';

import ContextTypes from './BpkBarchartContextTypes';

export const ANIMATE_PROPS = {
  calcMode: 'spline',
  keyTimes: '0; 1',
  keySplines: '.1,.8,.2,1',
  fill: 'freeze',
};

/* eslint-disable max-len */
// Taken from https://github.com/Modernizr/Modernizr/blob/7db55bbfa9de67289892b94b60bf2f088c11d669/feature-detects/svg/smil.js
/* eslint-enable */
const canAnimate =
  document &&
  !!document.createElementNS &&
  /SVGAnimate/.test(
    {}.toString.call(
      document.createElementNS('http://www.w3.org/2000/svg', 'animate'),
    ),
  );

const getYPos = (point, { yScaler, yScaleDataKey, maxYValue }) =>
  yScaler(Math.min(point[yScaleDataKey], maxYValue));

const getBarHeight = (
  point,
  { height, margin, yScaler, yScaleDataKey, maxYValue },
) => {
  const barHeight =
    height -
    margin.top -
    margin.bottom -
    getYPos(point, { yScaler, yScaleDataKey, maxYValue });
  return Math.max(barHeight, 0);
};

const getPathDescription = (width, height, x, y, radius, isOutlier = false) => {
  const path = d3Path();
  path.moveTo(x, y);

  if (isOutlier) {
    path.lineTo(x + width, y);
  } else {
    path.arcTo(x + width, y, x + width, y + height, radius);
  }

  path.arcTo(x + width, y + height, x, y + height, radius);
  path.arcTo(x, y + height, x, y, radius);

  if (isOutlier) {
    path.lineTo(x, y);
  } else {
    path.arcTo(x, y, x + width, y, radius);
  }
  return path.toString();
};

const isOutlier = (point, { yScaleDataKey, maxYValue }) =>
  point[yScaleDataKey] > maxYValue;

const BpkBarchartBars = (
  { animate, padding, rounded, onClick, animateDuration, ...rest },
  context,
) => {
  const { data, height, margin, xScaleDataKey, yScaleDataKey, xScaler } = context;
  xScaler.padding(padding);

  const shouldAnimate = animate && canAnimate;
  const barWidth = xScaler.bandwidth();
  const roundedAmount = rounded ? barWidth / 2 : 0;
  const heights = data.map(point => getBarHeight(point, context));
  const minHeight = Math.min(...heights);

  return (
    <g className="bpk-barchart__bars-group">
      {data.map((point, i) => {
        const x = xScaler(point[xScaleDataKey]);
        const initialD = shouldAnimate
          ? getPathDescription(
              barWidth,
              minHeight,
              x,
              height - margin.bottom - margin.top - minHeight,
              roundedAmount,
              isOutlier(point, context),
            )
          : null;
        const finalD = getPathDescription(
          barWidth,
          heights[i],
          x,
          getYPos(point, context),
          roundedAmount,
          isOutlier(point, context),
        );
        return (
          <g className="bpk-barchart__bar-group" key={`bar${i.toString()}`}>
            <title>{`${point[xScaleDataKey]} - ${point[yScaleDataKey]}`}</title>
            <path
              className={`bpk-barchart__bar ${isOutlier(point, context) ? 'bpk-barchart__bar--outlier' : ''}`}
              d={shouldAnimate ? initialD : finalD}
              onClick={e => onClick(e, point)}
              {...rest}
            >
              {shouldAnimate
                ? <animate
                  attributeName="d"
                  from={initialD}
                  to={finalD}
                  dur={animateDuration}
                  {...ANIMATE_PROPS}
                />
                : null}
            </path>
          </g>
        );
      })}
    </g>
  );
};

BpkBarchartBars.propTypes = {
  padding: PropTypes.number,
  rounded: PropTypes.bool,
  onClick: PropTypes.func,
  animate: PropTypes.bool,
  animateDuration: PropTypes.string,
};

BpkBarchartBars.defaultProps = {
  padding: 0.35,
  rounded: true,
  onClick: () => {},
  animate: false,
  animateDuration: durationBase,
};

BpkBarchartBars.contextTypes = ContextTypes;

export default BpkBarchartBars;
