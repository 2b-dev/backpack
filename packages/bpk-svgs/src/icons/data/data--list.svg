<svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24"><path d="M6 8H2V4h4v4zm0 2H2v4h4v-4zm0 6H2v4h4v-4zM22 4H8v4h14V4zm-2 6H8v4h12v-4zm-4 6H8v4h8v-4z"/></svg>