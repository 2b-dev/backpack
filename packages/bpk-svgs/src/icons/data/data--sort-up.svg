<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M18.79 8.6L12 3 5.21 8.6A.84.84 0 0 0 5.8 10h12.29a.85.85 0 0 0 .7-1.4"/></svg>