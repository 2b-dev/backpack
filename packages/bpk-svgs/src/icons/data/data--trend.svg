<svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24"><path d="M20.819 5.023l-5.177 1.069c-.693.143-.871 1.008-.322 1.558l.95.95-2.994 2.994H8.063c-.398 0-.779.158-1.061.439L2.44 16.596a1.5 1.5 0 1 0 2.122 2.121l4.123-4.123h5.213c.398 0 .779-.158 1.061-.439l3.434-3.434.959.959c.549.55 1.415.371 1.558-.322l1.069-5.177c.137-.674-.486-1.297-1.16-1.158z"/></svg>