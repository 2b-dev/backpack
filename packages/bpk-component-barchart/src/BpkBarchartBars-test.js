import React from 'react';
import renderer from 'react-test-renderer';
import { scaleLinear, scaleBand } from 'd3-scale';
import BpkBarchartBars from './BpkBarchartBars';
import data from '../data.json';

const margin = {
  top: 10,
  right: 20,
  bottom: 30,
  left: 40,
};
const continentCountries = data.continentCountries;
const size = 200;
const yScale = scaleLinear().domain([0, 50]).range([0, size]);
const xScale = scaleBand()
  .domain(continentCountries.map(d => d.continent)).range([0, size]);

describe('BpkBarchartBars', () => {
  it('should render correctly', () => {
    const tree = renderer.create(
      <BpkBarchartBars
        margin={margin}
        xScale={xScale}
        yScale={yScale}
        xScaleDataKey="continent"
        yScaleDataKey="countries"
        maxYValue={50}
        width={size}
        height={size}
        data={continentCountries}
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render with "rounded" prop set to "false"', () => {
    const tree = renderer.create(
      <BpkBarchartBars
        margin={margin}
        xScale={xScale}
        yScale={yScale}
        xScaleDataKey="continent"
        yScaleDataKey="countries"
        maxYValue={50}
        width={size}
        height={size}
        data={continentCountries}
        rounded={false}
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly with "padding" prop', () => {
    const tree = renderer.create(
      <BpkBarchartBars
        margin={margin}
        xScale={xScale}
        yScale={yScale}
        xScaleDataKey="continent"
        yScaleDataKey="countries"
        maxYValue={50}
        width={size}
        height={size}
        data={continentCountries}
        padding={0}
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
