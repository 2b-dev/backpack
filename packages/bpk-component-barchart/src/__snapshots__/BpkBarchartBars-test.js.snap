// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BpkBarchartBars should render correctly 1`] = `
<g>
  <g
    className="bpk-barchart__bar-group"
  >
    <title>
      Jan - 305
    </title>
    <rect
      aria-label="Jan - 305"
      className="bpk-barchart__bar bpk-barchart__bar--outlier"
      height={143}
      onClick={undefined}
      onKeyDown={undefined}
      role={undefined}
      rx={3}
      ry={3}
      tabIndex={undefined}
      width={200}
      x={5.668016194331983}
      y={17}
    />
  </g>
  <g
    className="bpk-barchart__bar-group"
  >
    <title>
      Feb - 348
    </title>
    <rect
      aria-label="Feb - 348"
      className="bpk-barchart__bar bpk-barchart__bar--outlier"
      height={143}
      onClick={undefined}
      onKeyDown={undefined}
      role={undefined}
      rx={3}
      ry={3}
      tabIndex={undefined}
      width={200}
      x={21.86234817813765}
      y={17}
    />
  </g>
  <g
    className="bpk-barchart__bar-group"
  >
    <title>
      Mar - 418
    </title>
    <rect
      aria-label="Mar - 418"
      className="bpk-barchart__bar bpk-barchart__bar--outlier"
      height={143}
      onClick={undefined}
      onKeyDown={undefined}
      role={undefined}
      rx={3}
      ry={3}
      tabIndex={undefined}
      width={200}
      x={38.05668016194332}
      y={17}
    />
  </g>
  <g
    className="bpk-barchart__bar-group"
  >
    <title>
      Apr - 448
    </title>
    <rect
      aria-label="Apr - 448"
      className="bpk-barchart__bar bpk-barchart__bar--outlier"
      height={143}
      onClick={undefined}
      onKeyDown={undefined}
      role={undefined}
      rx={3}
      ry={3}
      tabIndex={undefined}
      width={200}
      x={54.25101214574899}
      y={17}
    />
  </g>
  <g
    className="bpk-barchart__bar-group"
  >
    <title>
      May - 344
    </title>
    <rect
      aria-label="May - 344"
      className="bpk-barchart__bar bpk-barchart__bar--outlier"
      height={143}
      onClick={undefined}
      onKeyDown={undefined}
      role={undefined}
      rx={3}
      ry={3}
      tabIndex={undefined}
      width={200}
      x={70.44534412955466}
      y={17}
    />
  </g>
  <g
    className="bpk-barchart__bar-group"
  >
    <title>
      Jun - 322
    </title>
    <rect
      aria-label="Jun - 322"
      className="bpk-barchart__bar bpk-barchart__bar--outlier"
      height={143}
      onClick={undefined}
      onKeyDown={undefined}
      role={undefined}
      rx={3}
      ry={3}
      tabIndex={undefined}
      width={200}
      x={86.63967611336032}
      y={17}
    />
  </g>
  <g
    className="bpk-barchart__bar-group"
  >
    <title>
      Jul - 340
    </title>
    <rect
      aria-label="Jul - 340"
      className="bpk-barchart__bar bpk-barchart__bar--outlier"
      height={143}
      onClick={undefined}
      onKeyDown={undefined}
      role={undefined}
      rx={3}
      ry={3}
      tabIndex={undefined}
      width={200}
      x={102.83400809716599}
      y={17}
    />
  </g>
  <g
    className="bpk-barchart__bar-group"
  >
    <title>
      Aug - 327
    </title>
    <rect
      aria-label="Aug - 327"
      className="bpk-barchart__bar bpk-barchart__bar--outlier"
      height={143}
      onClick={undefined}
      onKeyDown={undefined}
      role={undefined}
      rx={3}
      ry={3}
      tabIndex={undefined}
      width={200}
      x={119.02834008097166}
      y={17}
    />
  </g>
  <g
    className="bpk-barchart__bar-group"
  >
    <title>
      Sep - 444
    </title>
    <rect
      aria-label="Sep - 444"
      className="bpk-barchart__bar bpk-barchart__bar--outlier"
      height={143}
      onClick={undefined}
      onKeyDown={undefined}
      role={undefined}
      rx={3}
      ry={3}
      tabIndex={undefined}
      width={200}
      x={135.22267206477733}
      y={17}
    />
  </g>
  <g
    className="bpk-barchart__bar-group"
  >
    <title>
      Oct - 422
    </title>
    <rect
      aria-label="Oct - 422"
      className="bpk-barchart__bar bpk-barchart__bar--outlier"
      height={143}
      onClick={undefined}
      onKeyDown={undefined}
      role={undefined}
      rx={3}
      ry={3}
      tabIndex={undefined}
      width={200}
      x={151.417004048583}
      y={17}
    />
  </g>
  <g
    className="bpk-barchart__bar-group"
  >
    <title>
      Nov - 344
    </title>
    <rect
      aria-label="Nov - 344"
      className="bpk-barchart__bar bpk-barchart__bar--outlier"
      height={143}
      onClick={undefined}
      onKeyDown={undefined}
      role={undefined}
      rx={3}
      ry={3}
      tabIndex={undefined}
      width={200}
      x={167.61133603238866}
      y={17}
    />
  </g>
  <g
    className="bpk-barchart__bar-group"
  >
    <title>
      Dec - 301
    </title>
    <rect
      aria-label="Dec - 301"
      className="bpk-barchart__bar bpk-barchart__bar--outlier"
      height={143}
      onClick={undefined}
      onKeyDown={undefined}
      role={undefined}
      rx={3}
      ry={3}
      tabIndex={undefined}
      width={200}
      x={183.80566801619432}
      y={17}
    />
  </g>
</g>
`;

exports[`BpkBarchartBars should render correctly with "onBarClick" prop 1`] = `
<g>
  <g
    className="bpk-barchart__bar-group"
  >
    <title>
      Jan - 305
    </title>
    <rect
      aria-label="Jan - 305"
      className="bpk-barchart__bar bpk-barchart__bar--outlier bpk-barchart__bar--interactive"
      height={143}
      onClick={[Function]}
      onKeyDown={[Function]}
      padding={0}
      role="button"
      rx={3}
      ry={3}
      tabIndex={0}
      width={200}
      x={5.668016194331983}
      y={17}
    />
  </g>
  <g
    className="bpk-barchart__bar-group"
  >
    <title>
      Feb - 348
    </title>
    <rect
      aria-label="Feb - 348"
      className="bpk-barchart__bar bpk-barchart__bar--outlier bpk-barchart__bar--interactive"
      height={143}
      onClick={[Function]}
      onKeyDown={[Function]}
      padding={0}
      role="button"
      rx={3}
      ry={3}
      tabIndex={0}
      width={200}
      x={21.86234817813765}
      y={17}
    />
  </g>
  <g
    className="bpk-barchart__bar-group"
  >
    <title>
      Mar - 418
    </title>
    <rect
      aria-label="Mar - 418"
      className="bpk-barchart__bar bpk-barchart__bar--outlier bpk-barchart__bar--interactive"
      height={143}
      onClick={[Function]}
      onKeyDown={[Function]}
      padding={0}
      role="button"
      rx={3}
      ry={3}
      tabIndex={0}
      width={200}
      x={38.05668016194332}
      y={17}
    />
  </g>
  <g
    className="bpk-barchart__bar-group"
  >
    <title>
      Apr - 448
    </title>
    <rect
      aria-label="Apr - 448"
      className="bpk-barchart__bar bpk-barchart__bar--outlier bpk-barchart__bar--interactive"
      height={143}
      onClick={[Function]}
      onKeyDown={[Function]}
      padding={0}
      role="button"
      rx={3}
      ry={3}
      tabIndex={0}
      width={200}
      x={54.25101214574899}
      y={17}
    />
  </g>
  <g
    className="bpk-barchart__bar-group"
  >
    <title>
      May - 344
    </title>
    <rect
      aria-label="May - 344"
      className="bpk-barchart__bar bpk-barchart__bar--outlier bpk-barchart__bar--interactive"
      height={143}
      onClick={[Function]}
      onKeyDown={[Function]}
      padding={0}
      role="button"
      rx={3}
      ry={3}
      tabIndex={0}
      width={200}
      x={70.44534412955466}
      y={17}
    />
  </g>
  <g
    className="bpk-barchart__bar-group"
  >
    <title>
      Jun - 322
    </title>
    <rect
      aria-label="Jun - 322"
      className="bpk-barchart__bar bpk-barchart__bar--outlier bpk-barchart__bar--interactive"
      height={143}
      onClick={[Function]}
      onKeyDown={[Function]}
      padding={0}
      role="button"
      rx={3}
      ry={3}
      tabIndex={0}
      width={200}
      x={86.63967611336032}
      y={17}
    />
  </g>
  <g
    className="bpk-barchart__bar-group"
  >
    <title>
      Jul - 340
    </title>
    <rect
      aria-label="Jul - 340"
      className="bpk-barchart__bar bpk-barchart__bar--outlier bpk-barchart__bar--interactive"
      height={143}
      onClick={[Function]}
      onKeyDown={[Function]}
      padding={0}
      role="button"
      rx={3}
      ry={3}
      tabIndex={0}
      width={200}
      x={102.83400809716599}
      y={17}
    />
  </g>
  <g
    className="bpk-barchart__bar-group"
  >
    <title>
      Aug - 327
    </title>
    <rect
      aria-label="Aug - 327"
      className="bpk-barchart__bar bpk-barchart__bar--outlier bpk-barchart__bar--interactive"
      height={143}
      onClick={[Function]}
      onKeyDown={[Function]}
      padding={0}
      role="button"
      rx={3}
      ry={3}
      tabIndex={0}
      width={200}
      x={119.02834008097166}
      y={17}
    />
  </g>
  <g
    className="bpk-barchart__bar-group"
  >
    <title>
      Sep - 444
    </title>
    <rect
      aria-label="Sep - 444"
      className="bpk-barchart__bar bpk-barchart__bar--outlier bpk-barchart__bar--interactive"
      height={143}
      onClick={[Function]}
      onKeyDown={[Function]}
      padding={0}
      role="button"
      rx={3}
      ry={3}
      tabIndex={0}
      width={200}
      x={135.22267206477733}
      y={17}
    />
  </g>
  <g
    className="bpk-barchart__bar-group"
  >
    <title>
      Oct - 422
    </title>
    <rect
      aria-label="Oct - 422"
      className="bpk-barchart__bar bpk-barchart__bar--outlier bpk-barchart__bar--interactive"
      height={143}
      onClick={[Function]}
      onKeyDown={[Function]}
      padding={0}
      role="button"
      rx={3}
      ry={3}
      tabIndex={0}
      width={200}
      x={151.417004048583}
      y={17}
    />
  </g>
  <g
    className="bpk-barchart__bar-group"
  >
    <title>
      Nov - 344
    </title>
    <rect
      aria-label="Nov - 344"
      className="bpk-barchart__bar bpk-barchart__bar--outlier bpk-barchart__bar--interactive"
      height={143}
      onClick={[Function]}
      onKeyDown={[Function]}
      padding={0}
      role="button"
      rx={3}
      ry={3}
      tabIndex={0}
      width={200}
      x={167.61133603238866}
      y={17}
    />
  </g>
  <g
    className="bpk-barchart__bar-group"
  >
    <title>
      Dec - 301
    </title>
    <rect
      aria-label="Dec - 301"
      className="bpk-barchart__bar bpk-barchart__bar--outlier bpk-barchart__bar--interactive"
      height={143}
      onClick={[Function]}
      onKeyDown={[Function]}
      padding={0}
      role="button"
      rx={3}
      ry={3}
      tabIndex={0}
      width={200}
      x={183.80566801619432}
      y={17}
    />
  </g>
</g>
`;

exports[`BpkBarchartBars should render correctly with "padding" prop 1`] = `
<g>
  <g
    className="bpk-barchart__bar-group"
  >
    <title>
      Jan - 305
    </title>
    <rect
      aria-label="Jan - 305"
      className="bpk-barchart__bar bpk-barchart__bar--outlier"
      height={143}
      onClick={undefined}
      onKeyDown={undefined}
      padding={0}
      role={undefined}
      rx={3}
      ry={3}
      tabIndex={undefined}
      width={200}
      x={5.668016194331983}
      y={17}
    />
  </g>
  <g
    className="bpk-barchart__bar-group"
  >
    <title>
      Feb - 348
    </title>
    <rect
      aria-label="Feb - 348"
      className="bpk-barchart__bar bpk-barchart__bar--outlier"
      height={143}
      onClick={undefined}
      onKeyDown={undefined}
      padding={0}
      role={undefined}
      rx={3}
      ry={3}
      tabIndex={undefined}
      width={200}
      x={21.86234817813765}
      y={17}
    />
  </g>
  <g
    className="bpk-barchart__bar-group"
  >
    <title>
      Mar - 418
    </title>
    <rect
      aria-label="Mar - 418"
      className="bpk-barchart__bar bpk-barchart__bar--outlier"
      height={143}
      onClick={undefined}
      onKeyDown={undefined}
      padding={0}
      role={undefined}
      rx={3}
      ry={3}
      tabIndex={undefined}
      width={200}
      x={38.05668016194332}
      y={17}
    />
  </g>
  <g
    className="bpk-barchart__bar-group"
  >
    <title>
      Apr - 448
    </title>
    <rect
      aria-label="Apr - 448"
      className="bpk-barchart__bar bpk-barchart__bar--outlier"
      height={143}
      onClick={undefined}
      onKeyDown={undefined}
      padding={0}
      role={undefined}
      rx={3}
      ry={3}
      tabIndex={undefined}
      width={200}
      x={54.25101214574899}
      y={17}
    />
  </g>
  <g
    className="bpk-barchart__bar-group"
  >
    <title>
      May - 344
    </title>
    <rect
      aria-label="May - 344"
      className="bpk-barchart__bar bpk-barchart__bar--outlier"
      height={143}
      onClick={undefined}
      onKeyDown={undefined}
      padding={0}
      role={undefined}
      rx={3}
      ry={3}
      tabIndex={undefined}
      width={200}
      x={70.44534412955466}
      y={17}
    />
  </g>
  <g
    className="bpk-barchart__bar-group"
  >
    <title>
      Jun - 322
    </title>
    <rect
      aria-label="Jun - 322"
      className="bpk-barchart__bar bpk-barchart__bar--outlier"
      height={143}
      onClick={undefined}
      onKeyDown={undefined}
      padding={0}
      role={undefined}
      rx={3}
      ry={3}
      tabIndex={undefined}
      width={200}
      x={86.63967611336032}
      y={17}
    />
  </g>
  <g
    className="bpk-barchart__bar-group"
  >
    <title>
      Jul - 340
    </title>
    <rect
      aria-label="Jul - 340"
      className="bpk-barchart__bar bpk-barchart__bar--outlier"
      height={143}
      onClick={undefined}
      onKeyDown={undefined}
      padding={0}
      role={undefined}
      rx={3}
      ry={3}
      tabIndex={undefined}
      width={200}
      x={102.83400809716599}
      y={17}
    />
  </g>
  <g
    className="bpk-barchart__bar-group"
  >
    <title>
      Aug - 327
    </title>
    <rect
      aria-label="Aug - 327"
      className="bpk-barchart__bar bpk-barchart__bar--outlier"
      height={143}
      onClick={undefined}
      onKeyDown={undefined}
      padding={0}
      role={undefined}
      rx={3}
      ry={3}
      tabIndex={undefined}
      width={200}
      x={119.02834008097166}
      y={17}
    />
  </g>
  <g
    className="bpk-barchart__bar-group"
  >
    <title>
      Sep - 444
    </title>
    <rect
      aria-label="Sep - 444"
      className="bpk-barchart__bar bpk-barchart__bar--outlier"
      height={143}
      onClick={undefined}
      onKeyDown={undefined}
      padding={0}
      role={undefined}
      rx={3}
      ry={3}
      tabIndex={undefined}
      width={200}
      x={135.22267206477733}
      y={17}
    />
  </g>
  <g
    className="bpk-barchart__bar-group"
  >
    <title>
      Oct - 422
    </title>
    <rect
      aria-label="Oct - 422"
      className="bpk-barchart__bar bpk-barchart__bar--outlier"
      height={143}
      onClick={undefined}
      onKeyDown={undefined}
      padding={0}
      role={undefined}
      rx={3}
      ry={3}
      tabIndex={undefined}
      width={200}
      x={151.417004048583}
      y={17}
    />
  </g>
  <g
    className="bpk-barchart__bar-group"
  >
    <title>
      Nov - 344
    </title>
    <rect
      aria-label="Nov - 344"
      className="bpk-barchart__bar bpk-barchart__bar--outlier"
      height={143}
      onClick={undefined}
      onKeyDown={undefined}
      padding={0}
      role={undefined}
      rx={3}
      ry={3}
      tabIndex={undefined}
      width={200}
      x={167.61133603238866}
      y={17}
    />
  </g>
  <g
    className="bpk-barchart__bar-group"
  >
    <title>
      Dec - 301
    </title>
    <rect
      aria-label="Dec - 301"
      className="bpk-barchart__bar bpk-barchart__bar--outlier"
      height={143}
      onClick={undefined}
      onKeyDown={undefined}
      padding={0}
      role={undefined}
      rx={3}
      ry={3}
      tabIndex={undefined}
      width={200}
      x={183.80566801619432}
      y={17}
    />
  </g>
</g>
`;

exports[`BpkBarchartBars should render with "rounded" prop set to "false" 1`] = `
<g>
  <g
    className="bpk-barchart__bar-group"
  >
    <title>
      Jan - 305
    </title>
    <rect
      aria-label="Jan - 305"
      className="bpk-barchart__bar bpk-barchart__bar--outlier"
      height={143}
      onClick={undefined}
      onKeyDown={undefined}
      role={undefined}
      rounded={false}
      rx={3}
      ry={3}
      tabIndex={undefined}
      width={200}
      x={5.668016194331983}
      y={17}
    />
  </g>
  <g
    className="bpk-barchart__bar-group"
  >
    <title>
      Feb - 348
    </title>
    <rect
      aria-label="Feb - 348"
      className="bpk-barchart__bar bpk-barchart__bar--outlier"
      height={143}
      onClick={undefined}
      onKeyDown={undefined}
      role={undefined}
      rounded={false}
      rx={3}
      ry={3}
      tabIndex={undefined}
      width={200}
      x={21.86234817813765}
      y={17}
    />
  </g>
  <g
    className="bpk-barchart__bar-group"
  >
    <title>
      Mar - 418
    </title>
    <rect
      aria-label="Mar - 418"
      className="bpk-barchart__bar bpk-barchart__bar--outlier"
      height={143}
      onClick={undefined}
      onKeyDown={undefined}
      role={undefined}
      rounded={false}
      rx={3}
      ry={3}
      tabIndex={undefined}
      width={200}
      x={38.05668016194332}
      y={17}
    />
  </g>
  <g
    className="bpk-barchart__bar-group"
  >
    <title>
      Apr - 448
    </title>
    <rect
      aria-label="Apr - 448"
      className="bpk-barchart__bar bpk-barchart__bar--outlier"
      height={143}
      onClick={undefined}
      onKeyDown={undefined}
      role={undefined}
      rounded={false}
      rx={3}
      ry={3}
      tabIndex={undefined}
      width={200}
      x={54.25101214574899}
      y={17}
    />
  </g>
  <g
    className="bpk-barchart__bar-group"
  >
    <title>
      May - 344
    </title>
    <rect
      aria-label="May - 344"
      className="bpk-barchart__bar bpk-barchart__bar--outlier"
      height={143}
      onClick={undefined}
      onKeyDown={undefined}
      role={undefined}
      rounded={false}
      rx={3}
      ry={3}
      tabIndex={undefined}
      width={200}
      x={70.44534412955466}
      y={17}
    />
  </g>
  <g
    className="bpk-barchart__bar-group"
  >
    <title>
      Jun - 322
    </title>
    <rect
      aria-label="Jun - 322"
      className="bpk-barchart__bar bpk-barchart__bar--outlier"
      height={143}
      onClick={undefined}
      onKeyDown={undefined}
      role={undefined}
      rounded={false}
      rx={3}
      ry={3}
      tabIndex={undefined}
      width={200}
      x={86.63967611336032}
      y={17}
    />
  </g>
  <g
    className="bpk-barchart__bar-group"
  >
    <title>
      Jul - 340
    </title>
    <rect
      aria-label="Jul - 340"
      className="bpk-barchart__bar bpk-barchart__bar--outlier"
      height={143}
      onClick={undefined}
      onKeyDown={undefined}
      role={undefined}
      rounded={false}
      rx={3}
      ry={3}
      tabIndex={undefined}
      width={200}
      x={102.83400809716599}
      y={17}
    />
  </g>
  <g
    className="bpk-barchart__bar-group"
  >
    <title>
      Aug - 327
    </title>
    <rect
      aria-label="Aug - 327"
      className="bpk-barchart__bar bpk-barchart__bar--outlier"
      height={143}
      onClick={undefined}
      onKeyDown={undefined}
      role={undefined}
      rounded={false}
      rx={3}
      ry={3}
      tabIndex={undefined}
      width={200}
      x={119.02834008097166}
      y={17}
    />
  </g>
  <g
    className="bpk-barchart__bar-group"
  >
    <title>
      Sep - 444
    </title>
    <rect
      aria-label="Sep - 444"
      className="bpk-barchart__bar bpk-barchart__bar--outlier"
      height={143}
      onClick={undefined}
      onKeyDown={undefined}
      role={undefined}
      rounded={false}
      rx={3}
      ry={3}
      tabIndex={undefined}
      width={200}
      x={135.22267206477733}
      y={17}
    />
  </g>
  <g
    className="bpk-barchart__bar-group"
  >
    <title>
      Oct - 422
    </title>
    <rect
      aria-label="Oct - 422"
      className="bpk-barchart__bar bpk-barchart__bar--outlier"
      height={143}
      onClick={undefined}
      onKeyDown={undefined}
      role={undefined}
      rounded={false}
      rx={3}
      ry={3}
      tabIndex={undefined}
      width={200}
      x={151.417004048583}
      y={17}
    />
  </g>
  <g
    className="bpk-barchart__bar-group"
  >
    <title>
      Nov - 344
    </title>
    <rect
      aria-label="Nov - 344"
      className="bpk-barchart__bar bpk-barchart__bar--outlier"
      height={143}
      onClick={undefined}
      onKeyDown={undefined}
      role={undefined}
      rounded={false}
      rx={3}
      ry={3}
      tabIndex={undefined}
      width={200}
      x={167.61133603238866}
      y={17}
    />
  </g>
  <g
    className="bpk-barchart__bar-group"
  >
    <title>
      Dec - 301
    </title>
    <rect
      aria-label="Dec - 301"
      className="bpk-barchart__bar bpk-barchart__bar--outlier"
      height={143}
      onClick={undefined}
      onKeyDown={undefined}
      role={undefined}
      rounded={false}
      rx={3}
      ry={3}
      tabIndex={undefined}
      width={200}
      x={183.80566801619432}
      y={17}
    />
  </g>
</g>
`;
